Sources/AsyncHTTPClient/Body+Encodable.swift

```diff
@@ -0,0 +1,31 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2018-2019 Swift Server Working Group and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+
+import Foundation
+
+extension HTTPClient.Body {
+    
+    public static func encodable<T : Encodable>(_ encodable: T, encoder: Encoder = JSONEncoder()) throws -> HTTPClient.Body {
+        return HTTPClient.Body.data(try encoder.encode(encodable))
+    }
+    
+}
+
+public protocol Encoder {
+    func encode<T>(_ value: T) throws -> Data where T : Encodable
```

Comment:
In Foundation `JSONEncoder` deliberately doesn't implement this method via a protocol, it just has a method for it with no protocol involved: https://github.com/apple/swift-corelibs-foundation/blob/2195121bf063e2db7c71080fcae7e76e8a43950e/Foundation/JSONEncoder.swift#L234

AFAIK this was a deliberate design choice in Foundation.

It seems to me that if this protocol is going to exist it really belongs in Foundation.

Reply:
I agree that this belongs in Foundation.  I simply added it here to support the interchange of encoders and decoders.  However, I also agree with @Lukasa that this functionality is probably not needed.  If we want I can remove these protocol and the encoder and decoder interchange api.

Reply:
With that being said I implemented this functionality with two things in mind.
1.  The ability to inject coders for other protocols, e.g., XML, YAML.
2.  The ability to inject a customized json coder, e.g., a JSONDecoder where keyDecodingStrategy = .convertFromSnakeCase.

Again this may be overkill for the API but I felt it was a nice optional parameter to have.

Reply:
@ianpartridge what do you think about my comments above?  Should I remove this functionality?

Reply:
I think so, personally.